src/studio/src/designer/frontend/packages/text-editor/src/mutations.ts

```diff
@@ -19,18 +19,27 @@ export const removeTextEntry = (
 };
 
 export const upsertTextEntry = (
-  translations: TextResourceFile,
+  resourceFile: TextResourceFile,
   newEntry: TextResourceEntry
 ): TextResourceFile => {
-  const updatedTranslations = deepCopy(translations);
-  const existingEntryIndex = updatedTranslations.resources.findIndex(
-    (entry) => entry.id === newEntry.id
-  );
-  if (updatedTranslations.resources[existingEntryIndex]) {
-    updatedTranslations.resources[existingEntryIndex].value = newEntry.value;
-    updatedTranslations.resources[existingEntryIndex].variables = newEntry.variables;
+  const updatedFile = deepCopy(resourceFile);
+  const existingEntryIndex = updatedFile.resources.findIndex((e) => e.id === newEntry.id);
+  if (updatedFile.resources[existingEntryIndex]) {
+    updatedFile.resources[existingEntryIndex].value = newEntry.value;
+    updatedFile.resources[existingEntryIndex].variables = newEntry.variables;
   } else {
-    updatedTranslations.resources.push(newEntry);
+    updatedFile.resources.push(newEntry);
+  }
+  return updatedFile;
+};
+
+export const updateTextEntryId = (resourceFile: TextResourceFile, oldId: string, newId: string) => {
+  const updatedFile = deepCopy(resourceFile);
+  const existingEntryIndex = updatedFile.resources.findIndex((e) => e.id === oldId);
+  if (updatedFile.resources[existingEntryIndex]) {
+    updatedFile.resources[existingEntryIndex].id = newId;
+    return updatedFile;
+  } else {
+    throw Error("Can't find this text-id");
```

Comment:
Do we have error handling for this, or will the page crash?

Reply:
There is no need to throw an error, the API should be called regardless, as it might exist on other languages.

Reply:
Det er ingen logikk på backend som sørger for at man kan oppdatere en nøkkel på tvers av filer. Dette er heller ikke implementert i dagens løsning eller den koden som jeg arvet da jeg startet dette. Så funksjonelt er dette helt likt det som var der fra før.

Reply:
@nkylstad det er en sjekk på dette i forkant... så dette skal aldri skje. men om det skjer så ønsker jeg at vi crasher appen. for da har vi kommet i en state vi ikke skal komme til. Da har vi rett og slett introdusert en feil.

Reply:
👍 Burde sikkert få på plass et fungerende error boundary etter hvert sånn at brukeren ikke får bare en blank skjerm, men det blir en annen sak.